Dockerfile

```diff
@@ -1,32 +1,45 @@
 FROM node:8.15.0
 
-# Install VS Code's deps. These are the only two it seems we need.
 RUN apt-get update && apt-get install -y \
 	libxkbfile-dev \
 	libsecret-1-dev
 
-# Ensure latest yarn.
-RUN npm install -g yarn@1.13
-
 WORKDIR /src
 COPY . .
 
-# In the future, we can use https://github.com/yarnpkg/rfcs/pull/53 to make yarn use the node_modules
-# directly which should be fast as it is slow because it populates its own cache every time.
-RUN yarn && yarn task build:server:binary
+# This takes ages and always dies in the end. :(
+# RUN yarn --frozen-lockfile && yarn task build:server:binary
+
+# Make the debugging easier - and the rebuilds faster
+# and our life happier lets break up the build to sequential parts
+RUN yarn --frozen-lockfile 
+
+RUN yarn task vscode:install
+RUN yarn task build:copy-vscode
+RUN yarn task build:web
+RUN yarn task build:bootstrap-fork
+RUN yarn task build:default-extensions
+RUN yarn task build:server:bundle
+RUN yarn task build:app:browser
+RUN yarn task build:server:binary:package
```

Comment:
Tasks depend on prior tasks. Running in this way will likely increase total build time significantly.

Reply:
Yes @kylecarbs I think so too
Changing this to 'supposed' serial execution ruins the idea of tasks.ts file which clearly respects the dependencies.

The only benefit - that is to admit - there would be a Docker build layer for each step, allowing to cache single yarn steps

Reply:
That's a fair point. We can also optimize to cache previous tasks.

Reply:
We can't cache single yarn steps with this because the input to each step is the same source. So every time the source changes, everything will still be rebuilt, it won't get reused.

Reply:
I would like to also add this is very inefficient and doesn't really do what your PR is supposed to do.